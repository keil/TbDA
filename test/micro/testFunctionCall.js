// plain function invocation
function f(x, y) {
	return x+y;
}
var a1 = f(17, 4);
dumpValue(a1);

// function invocation via call without receiver
function g(x, y) {
	return x-y;
}
var a2 = g.call(null, 101, 1);
dumpValue(a2);

// function invocation via call with receiver
function h() {
	return this.x;
}
var a3 = h.call({ x: "hoohoo" });
dumpValue(a3);

// calling a primitve
// var d = "xxx".concat("yyy"); // works ok
var a4 = "".concat.call("xxx", "yyy");
dumpValue(a4);

// recursive calling
var a5 = "zzz".substring.call.call("".concat, "xxx", "yyy");
dumpValue(a5);

var a6 = Function.call.call("".concat, "xxx", "yyy");
dumpValue(a6);

var a7 = f.call.call("".concat, "xxx", "yyy");
dumpValue(a7);

var a8 = f.call.call.call("".concat, "xxx", "yyy");
dumpValue(a8);
