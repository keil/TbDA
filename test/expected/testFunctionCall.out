test/micro/testFunctionCall.js:24: [definite] Converting primitive string to object
test/micro/testFunctionCall.js:28: [definite] Converting primitive string to object
test/micro/testFunctionCall.js:28: [definite] Converting primitive string to object
test/micro/testFunctionCall.js:31: [definite] Converting primitive string to object
test/micro/testFunctionCall.js:34: [definite] Converting primitive string to object
test/micro/testFunctionCall.js:37: [definite] Converting primitive string to object
test/micro/testFunctionCall.js:12: [maybe] No object present
test/micro/testFunctionCall.js:24: [maybe] Converting primitive string to object
test/micro/testFunctionCall.js:28: [maybe] Converting primitive string to object
test/micro/testFunctionCall.js:31: [maybe] Converting primitive string to object
test/micro/testFunctionCall.js:34: [maybe] Converting primitive string to object
test/micro/testFunctionCall.js:37: [maybe] Converting primitive string to object
test/micro/testFunctionCall.js:6: [info] Abstract value: 21.0
test/micro/testFunctionCall.js:13: [info] Abstract value: 100.0
test/micro/testFunctionCall.js:20: [info] Abstract value: "hoohoo"
test/micro/testFunctionCall.js:25: [info] Abstract value: Str
test/micro/testFunctionCall.js:29: [info] Abstract value: Str
test/micro/testFunctionCall.js:32: [info] Abstract value: Str
test/micro/testFunctionCall.js:35: [info] Abstract value: Str
test/micro/testFunctionCall.js:38: [info] Abstract value: Str
